### PR TITLE
feat(bars): add categorical tick label truncation and rotation (#531)

### PR DESCRIPTION
**Pull request by samussiah**
## Summary

Adds configurable tick label truncation and rotation to the `bars` module for handling long category axis labels on country/site charts.

## Changes

- `scales.x.ticks.maxLength` - truncates category labels exceeding the specified character count, appending an ellipsis. The full label is shown in the tooltip on hover.
- `scales.x.ticks.rotation` - fixes the tick label rotation angle (0-90 degrees). When omitted, Chart.js applies its default auto-rotation.
- New `truncateLabel` utility (`src/bars/truncateLabel.js`)
- Validation in `validateSpec.js` for both new options
- Deep-merge support in `mergeSpec.js` and `updateSpec.js`
- Tooltip title callback injection in `buildTooltip.js` (shows full label when truncated)
- Documentation updated in `examples/docs/bars.md`
- 32 new tests across 3 test files (truncateLabel, getScales ticks/rotation, buildTooltip title, integration)

## Resolved Issues

- Resolves #531

### CODE REVIEW

File: src/bars/truncateLabel.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Truncate a label to a maximum character length, appending an ellipsis (…)
+ * when the label is shortened.
+ *
+ * @param {*} label - the tick label (coerced to string)
+ * @param {number|undefined|null} maxLength - maximum character length; falsy = no truncation
```
[the remark this answers is not in the document]
Comment (samussiah) [reply src/bars/truncateLabel.js]: Addressed in 1f65b48 — updated JSDoc to say 'positive integer; omit or pass undefined/null to skip truncation'.

File: src/bars/truncateLabel.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ * @returns {string} the (possibly truncated) label
+ */
+export default function truncateLabel(label, maxLength) {
+    if (label == null) return '';
+    const str = String(label);
+    if (!maxLength || str.length <= maxLength) return str;
+    return str.slice(0, maxLength - 1) + '\u2026';
```
[the remark this answers is not in the document]
Comment (samussiah) [reply src/bars/truncateLabel.js]: Addressed in 1f65b48 — added a Number.isFinite guard so negative/NaN values are treated as no-op. Added tests for both edge cases.

File: index.js [hunk 3]
```diff
@@ -25173,6 +25201,14 @@ var gsmViz = (() => {
     return { datasets, labels, nExcluded, nRowsExcluded };
   }
 
+  // src/bars/truncateLabel.js
+  function truncateLabel(label, maxLength) {
+    if (label == null) return "";
```
[the remark this answers is not in the document]
Comment (samussiah) [reply index.js:25237]: This is expected — the repo commits index.js as a generated bundle via a pre-commit hook. It is not hand-edited.
